ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java

```diff
@@ -59,121 +63,207 @@ public CCodeGenerator() {
         config.setClassForTemplateLoading(CCodeGenerator.class, "/templates");
     }
 
-    public Path getRequestOutputPath(final Request request) {
-        return Paths.get(outputDirectory + "/ds3_c_sdk/src/requests/" + RequestHelper.getNameRootUnderscores(request.getName()) + ".c");
+    public void setFileUtils(final FileUtils fileUtils) {
+        this.fileUtils = fileUtils;
     }
 
-    public Path getTypeOutputPath(final Type type) {
-        return Paths.get(outputDirectory + "/ds3_c_sdk/src/types/" + TypeHelper.getNameUnderscores(type.getName()) + ".h");
+    public void setSpec(final Ds3ApiSpec spec) {
+        this.spec = spec;
     }
 
     @Override
     public void generate(final Ds3ApiSpec spec, final FileUtils fileUtils, final Path destDir) throws IOException {
         this.spec = spec;
         this.fileUtils = fileUtils;
-        this.outputDirectory = destDir;
 
         try {
-            generateCommands();
-        } catch (final TemplateException e) {
+            generateDs3_H();
+            generateDs3_C();
+        } catch (final TemplateException | ParseException e) {
             e.printStackTrace();
         }
     }
 
-    private void generateCommands() throws IOException, TemplateException {
+    public void generateDs3_H() throws IOException, ParseException {
```

Comment:
My main comment here, is that for each template file, we should only call processTemplate on it once.  So basically should be changed is that, instead of processing a template after you create the model, you should create lists of the models, and then pass those to the template, and let it iterate through all the models to generate.